app/src/endpoints.c

```diff
@@ -139,6 +166,12 @@ int zmk_endpoints_send_report(uint16_t usage_page) {
         return send_keyboard_report();
     case HID_USAGE_CONSUMER:
         return send_consumer_report();
+    // FIXME: we should not and the usage page here, we should make sure
+    // the whole page gets here
+#if IS_ENABLED(CONFIG_ZMK_PLOVER_HID)
+    case (HID_USAGE_PLOVER & 0xFF):
```

Comment:
Odd this was necessary. can you explain what caused this that you hit it?

Reply:
The reason is in `app/include/zmk/events/keycode_state_changed.h` where the usage page is anded with `0xFF` before raising the event.

It seems like this is related to mod-taps/layer-taps somehow, since i tried removing that bitmask and that caused all mod-taps and layer-taps in my keymap to fail. My guess is that the high page bits are reused for something different in those behaviors, but I haven't researched it further.

I could add a `raw_usage_page` field to the event and use that instead, or those behaviors could be refactored, but I agree  that masking in the switch-statement isn't the solution.